src/data/fin.lean

```diff
@@ -62,6 +62,9 @@ We define the following operations:
 universe u
 open fin nat function
 
+/-- Projection for `fin` -/
+def fin.val {n} (x : fin n) : ℕ := subtype.val x
```

Comment:
@jcommelin what do you think of adding this back? I think you mentioned in #3955 that we should now prefer coercing to `nat`.